core/src/main/scala/com/microsoft/azure/synapse/ml/causal/DoubleMLEstimator.scala

```diff
@@ -313,13 +359,151 @@ class DoubleMLModel(val uid: String)
+    val predictionTreatedResidualCol =
+      DatasetExtensions.findUnusedColumnName("predictionTreatedResidual", dataset)
+    val rawPredictionTreatedResidualCol =
+      DatasetExtensions.findUnusedColumnName("rawPredictionTreatedResidual", dataset)
+    val probabilityTreatedResidualCol =
+      DatasetExtensions.findUnusedColumnName("probabilityTreatedResidual", dataset)
+    val dfTreated =
+      dfTransformedwithTreatedResidualModel.withColumnRenamed("prediction", predictionTreatedResidualCol)
+        .withColumnRenamed("rawPrediction", rawPredictionTreatedResidualCol)
+        .withColumnRenamed("probability", probabilityTreatedResidualCol)
+    val treatmentResidualVAUntreated =
+      new VectorAssembler()
+        .setInputCols(Array(probabilityUntreatedCol))
+        .setOutputCol("treatmentResidualVec")
+        .setHandleInvalid("skip")
+    val dfTransformedUntreated = treatmentResidualVAUntreated.transform(dfTreated)
+    val dfTransformedUntreatedResidualModel =
+      regressorModel.transform(dfTransformedUntreated).drop("treatmentResidualVec")
+    val predictionUntreatedResidualCol =
+      DatasetExtensions.findUnusedColumnName("predictionUntreatedResidual", dataset)
+    val rawPredictionUntreatedResidualCol =
+      DatasetExtensions.findUnusedColumnName("rawPredictionUntreatedResidual", dataset)
+    val probabilityUntreatedResidualCol =
+      DatasetExtensions.findUnusedColumnName("probabilityUntreatedResidual", dataset)
+    val dfProcessed =
+      dfTransformedUntreatedResidualModel.withColumnRenamed("prediction", predictionUntreatedResidualCol)
+        .withColumnRenamed("rawPrediction", rawPredictionUntreatedResidualCol)
+        .withColumnRenamed("probability", probabilityUntreatedResidualCol)
+
+    // a = (predictionOutcomeCol + predictionTreatedResidualCol)
+    // b = predictionTreatedResidualCol
+    // a is the unbiased predicted outcome if the individual gets treatment
+    // b is the unbiased predicted outcome if the individual does not get treatment
+    // (a - b)  is the individual treatment effect
+    val dfFinal =
+      dfProcessed.withColumn(iteColName,
+        col(predictionOutcomeCol) + col(predictionTreatedResidualCol) - col(predictionTreatedResidualCol)
+      )
+
+    val dfFinalCleaned = dfFinal.drop(
+      predictionTreatmentCol, rawPredictionTreatmentCol, probabilityTreatmentCol,
+      predictionOutcomeCol, rawPredictionOutcomeCol, probabilityOutcomeCol,
+      probabilityTreatedCol, probabilityUntreatedCol,
+      predictionTreatedResidualCol, rawPredictionTreatedResidualCol, probabilityTreatedResidualCol,
+      predictionUntreatedResidualCol, rawPredictionUntreatedResidualCol, probabilityUntreatedResidualCol
+    )
+
+    dfFinalCleaned
+  }
+
   @DeveloperApi
   override def transformSchema(schema: StructType): StructType =
-    StructType(schema.fields)
+    schema.add(getIteOutputCol, DoubleType)
+      .add(getIteStddevOutputCol, DoubleType)
```

Comment:
Does the transform function return just the ITEs? IIRC we were thinking of also returning the "causally-debiased" predictions right? Or maybe these are the same things in your implt. In particular the question is whether these are in the same "space" as the labels

Reply:
@mhamilton723, we can only return the estimated ITEs. Technically we can calculate the "causally-debiased" predictions easily, but in a few cases, you'd end up with predictions that clearly don't make sense (e.g. negative predictions where the prediction should be positive). 
@sarahshy please correct me if wrong.

Reply:
Based on Sarah's comments, " Jason Your comment on the doc, which is correct, simply returns the slope of the final residuals model (which is the ATE). So everyone's ITE is going to be the ATE. ", I'll close this PR.